apps/src/lib/bench_utils.rs

```diff
@@ -725,9 +744,126 @@ impl Client for BenchShell {
     where
         R: tendermint_rpc::SimpleRequest,
     {
-        Ok(R::Output::from(
-            tendermint_rpc::Response::from_string("MOCK RESPONSE").unwrap(),
-        ))
+        unimplemented!(
+            "Arbitrary queries are not implemented in the benchmarks context"
+        );
+    }
+
+    async fn block<H>(
+        &self,
+        height: H,
+    ) -> Result<tendermint_rpc::endpoint::block::Response, tendermint_rpc::Error>
+    where
+        H: Into<tendermint::block::Height> + Send,
+    {
+        // NOTE: atm this is only needed to query blocks at a specific height
+        // for masp transactions
+        let height = BlockHeight(height.into().into());
+
+        // Given the way we setup and run benchmarks, the masp transactions can
+        // only present in the last block, we can mock the previous
+        // responses with an empty set of transactions
+        let last_block_txs = if height
+            == self.inner.wl_storage.storage.get_last_block_height()
+        {
+            self.last_block_masp_txs.clone()
+        } else {
+            vec![]
+        };
+        Ok(tendermint_rpc::endpoint::block::Response {
+            block_id: tendermint::block::Id {
+                hash: tendermint::Hash::None,
+                part_set_header: tendermint::block::parts::Header::default(),
+            },
+            block: tendermint::block::Block::new(
+                tendermint::block::Header {
+                    version: tendermint::block::header::Version {
+                        block: 0,
+                        app: 0,
+                    },
+                    chain_id: self
+                        .inner
+                        .chain_id
+                        .to_string()
+                        .try_into()
+                        .unwrap(),
+                    height: 1u32.into(),
+                    time: tendermint::Time::now(),
+                    last_block_id: None,
+                    last_commit_hash: None,
+                    data_hash: None,
+                    validators_hash: tendermint::Hash::None,
+                    next_validators_hash: tendermint::Hash::None,
+                    consensus_hash: tendermint::Hash::None,
+                    app_hash: tendermint::AppHash::default(),
+                    last_results_hash: None,
+                    evidence_hash: None,
+                    proposer_address: tendermint::account::Id::new([0u8; 20]),
+                },
+                last_block_txs.into_iter().map(|tx| tx.to_bytes()).collect(),
+                tendermint::evidence::List::default(),
+                None,
+            )
+            .unwrap(),
+        })
+    }
+
+    async fn block_results<H>(
+        &self,
+        height: H,
+    ) -> Result<
+        tendermint_rpc::endpoint::block_results::Response,
+        tendermint_rpc::Error,
+    >
+    where
+        H: TryInto<namada::tendermint::block::Height> + Send,
```

Comment:
why are the different trait bound on H as opposed to how it's done in the block method? In fact, the logic for checking the latest block heeight seems like it should be the same but it's implemented two different ways